projects/step-core/src/lib/modules/basics/services/item-by-id-cache.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { Observable, of, tap } from 'rxjs';
+
+@Injectable()
+export abstract class ItemByIdCacheService<T> implements OnDestroy {
+  private itemsCache = new Map<string, T>();
+
+  getItem(id: string): Observable<T> {
+    if (this.itemsCache.has(id)) {
+      return of(this.itemsCache.get(id)!);
```

Comment:
what you can do here instead of using .has() and then .get() with ! assertion, you can say ```const existingItem = this.itemsCache.get(id);``` and then ```if (existingItem) { return of(existingItem); }```